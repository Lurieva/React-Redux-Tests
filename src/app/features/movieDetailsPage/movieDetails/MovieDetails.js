import React from 'react';

import './movieDetails.scss';

const MovieDetails = ({ movie }) => {
    const year = new Date(movie.release_date).getFullYear().toString();

    return (
        <div className="movie-details">
            <div className="movie-details__poster">
                <img src={movie.poster_path} 
                    alt={movie.title} />
            </div>
            <div className="movie-details__info">
                <div className="movie-details__info__title">{movie.title}</div>
                <div className="movie-details__info__tagline">{movie.tagline}</div>
                <div className="movie-details__info__year">{year}</div>
                <div className="movie-details__info__overview">{movie.overview}</div>
                <div className="movie-details__info__director">Director: </div>
                <div className="movie-details__info__director">Cast: </div>
            </div>
        </div>
    );
};

export default MovieDetails;