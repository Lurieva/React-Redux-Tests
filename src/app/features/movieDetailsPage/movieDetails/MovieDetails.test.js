import React from 'react';
import { shallow } from 'enzyme';
import MovieDetails from './MovieDetails';

describe('MovieDetails', () => {
    const movieMock = {
        id: 1,
        title: 'title',
        poster_path: 'poster_path',
        tagline: 'tagline',
        overview: 'overview',
        year: '12-01-2018'
    };

    it('should be defined', () => {
        expect(MovieDetails).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<MovieDetails movie={movieMock} />);
        expect(tree).toMatchSnapshot();
    });
});