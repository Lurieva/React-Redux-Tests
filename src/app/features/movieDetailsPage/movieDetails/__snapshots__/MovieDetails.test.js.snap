// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDetails
    movie={
      Object {
        "id": 1,
        "overview": "overview",
        "poster_path": "poster_path",
        "tagline": "tagline",
        "title": "title",
        "year": "12-01-2018",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-details__poster"
        >
          <img
            alt="title"
            src="poster_path"
          />
        </div>,
        <div
          className="movie-details__info"
        >
          <div
            className="movie-details__info__title"
          >
            title
          </div>
          <div
            className="movie-details__info__tagline"
          >
            tagline
          </div>
          <div
            className="movie-details__info__info"
          >
            <div
              className="movie-details__info__year"
            >
              NaN
            </div>
            <div
              className="movie-details__info__runtime"
            >
               min
            </div>
          </div>
          <div
            className="movie-details__info__overview"
          >
            overview
          </div>
        </div>,
      ],
      "className": "movie-details",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="title"
            src="poster_path"
          />,
          "className": "movie-details__poster",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "title",
            "src": "poster_path",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movie-details__info__title"
            >
              title
            </div>,
            <div
              className="movie-details__info__tagline"
            >
              tagline
            </div>,
            <div
              className="movie-details__info__info"
            >
              <div
                className="movie-details__info__year"
              >
                NaN
              </div>
              <div
                className="movie-details__info__runtime"
              >
                 min
              </div>
            </div>,
            <div
              className="movie-details__info__overview"
            >
              overview
            </div>,
          ],
          "className": "movie-details__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
              "className": "movie-details__info__title",
            },
            "ref": null,
            "rendered": "title",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "tagline",
              "className": "movie-details__info__tagline",
            },
            "ref": null,
            "rendered": "tagline",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="movie-details__info__year"
                >
                  NaN
                </div>,
                <div
                  className="movie-details__info__runtime"
                >
                   min
                </div>,
              ],
              "className": "movie-details__info__info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NaN",
                  "className": "movie-details__info__year",
                },
                "ref": null,
                "rendered": "NaN",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " min",
                  ],
                  "className": "movie-details__info__runtime",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " min",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "overview",
              "className": "movie-details__info__overview",
            },
            "ref": null,
            "rendered": "overview",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-details__poster"
          >
            <img
              alt="title"
              src="poster_path"
            />
          </div>,
          <div
            className="movie-details__info"
          >
            <div
              className="movie-details__info__title"
            >
              title
            </div>
            <div
              className="movie-details__info__tagline"
            >
              tagline
            </div>
            <div
              className="movie-details__info__info"
            >
              <div
                className="movie-details__info__year"
              >
                NaN
              </div>
              <div
                className="movie-details__info__runtime"
              >
                 min
              </div>
            </div>
            <div
              className="movie-details__info__overview"
            >
              overview
            </div>
          </div>,
        ],
        "className": "movie-details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="title"
              src="poster_path"
            />,
            "className": "movie-details__poster",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "title",
              "src": "poster_path",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movie-details__info__title"
              >
                title
              </div>,
              <div
                className="movie-details__info__tagline"
              >
                tagline
              </div>,
              <div
                className="movie-details__info__info"
              >
                <div
                  className="movie-details__info__year"
                >
                  NaN
                </div>
                <div
                  className="movie-details__info__runtime"
                >
                   min
                </div>
              </div>,
              <div
                className="movie-details__info__overview"
              >
                overview
              </div>,
            ],
            "className": "movie-details__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
                "className": "movie-details__info__title",
              },
              "ref": null,
              "rendered": "title",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "tagline",
                "className": "movie-details__info__tagline",
              },
              "ref": null,
              "rendered": "tagline",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="movie-details__info__year"
                  >
                    NaN
                  </div>,
                  <div
                    className="movie-details__info__runtime"
                  >
                     min
                  </div>,
                ],
                "className": "movie-details__info__info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "NaN",
                    "className": "movie-details__info__year",
                  },
                  "ref": null,
                  "rendered": "NaN",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " min",
                    ],
                    "className": "movie-details__info__runtime",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " min",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "overview",
                "className": "movie-details__info__overview",
              },
              "ref": null,
              "rendered": "overview",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
